lib/clevertap-api.js

```diff
@@ -24,10 +27,19 @@ var CleverTapAPI = function (CleverTapAccountId, CleverTapAccountPasscode) {
     this.accountId = CleverTapAccountId;
     this.accountPasscode = CleverTapAccountPasscode;
 
-    this.apiEndpoint = `https://${API_HOSTNAME}/${API_VERSION}`
+    this.apiEndpoint=findEndPoint(CleverTapEndPoint);
 
 };
 
+function findEndPoint(CleverTapEndPoint){
+    if(CleverTapEndPoint && (END_POINTS.includes(CleverTapEndPoint))){
+        this.endpoint=CleverTapEndPoint;
+    }else{
+        console.log("Please provide a valid API Endpoint");
```

Comment:
If a user doesn't specify `CleverTapEndPoint` this console log statement would be printed. `CleverTapEndPoint` is optional so we shouldn't be printing this.
Also in case of invalid endpoint throw a new Error rather than allowing the user to proceed. Thoughts?

Reply:
I had a discussion with this harsh,
He told that if a person is not passing the endpoint or pass some wrong still we should give him access with a info,
that he had entered a wrong endpoint.

Reply:
ok @abhishek-g-clevertap but then as already commented validate the region in the constructor and throw an error if needed - just like its done for account id and passcode that way all validations are consistent

Reply:
Need to discuss this with harsh, throwing an error is an option he discarded.
saying that the user should proceed unless he has entered the wrong region. we will inform showing a console message.

Reply:
if its optional log it in the constructor instead of throwing

Reply:
@pwilkniss Done.
Can you please once check and merge the changes.